mm2src/mm2_main/src/ext_api/one_inch/rpcs.rs

```diff
@@ -0,0 +1,148 @@
+use mm2_err_handle::prelude::*;
+use trading_api::one_inch_api::client::ApiClient;
+use trading_api::one_inch_api::types::{ClassicSwapCreateParams, ClassicSwapQuoteParams};
+
+// Default swap optional params:
+const INCLUDE_GAS: bool = true;
+const INCLUDE_PROTOCOLS: bool = true;
+const INCLUDE_TOKENS_INFO: bool = true;
+
+/// "1inch_v6_0_classic_swap_contract" rpc impl
+/// used to get contract address (for e.g. to approve funds)
+pub async fn one_inch_v6_0_classic_swap_contract_rpc(
+    _ctx: MmArc,
+    _req: AggregationContractRequest,
+) -> MmResult<String, ApiIntegrationRpcError> {
+    Ok(ApiClient::classic_swap_contract().to_owned())
+}
+
+/// "1inch_classic_swap_quote" rpc impl
+pub async fn one_inch_v6_0_classic_swap_quote_rpc(
+    ctx: MmArc,
+    req: ClassicSwapQuoteRequest,
+) -> MmResult<ClassicSwapResponse, ApiIntegrationRpcError> {
+    let (base, base_contract) = get_coin_for_one_inch(&ctx, &req.base).await?;
+    api_supports_coin(&base)?;
+    let (rel, rel_contract) = get_coin_for_one_inch(&ctx, &req.rel).await?;
+    let sell_amount = wei_from_big_decimal(&req.amount, base.decimals())
+        .mm_err(|err| ApiIntegrationRpcError::InvalidParam(err.to_string()))?;
+    let query_params = ClassicSwapQuoteParams::new(base_contract, rel_contract, sell_amount.to_string())
+        .with_fee(req.fee)
+        .with_protocols(req.protocols)
+        .with_gas_price(req.gas_price)
+        .with_complexity_level(req.complexity_level)
+        .with_parts(req.parts)
+        .with_main_route_parts(req.main_route_parts)
+        .with_gas_limit(req.gas_limit)
+        .with_include_tokens_info(if let Some(include_tokens_info) = req.include_tokens_info {
+            Some(include_tokens_info)
+        } else {
+            Some(INCLUDE_TOKENS_INFO)
+        })
+        .with_include_protocols(if let Some(include_protocols) = req.include_protocols {
+            Some(include_protocols)
+        } else {
+            Some(INCLUDE_PROTOCOLS)
+        })
+        .with_include_gas(if let Some(include_gas) = req.include_gas {
+            Some(include_gas)
+        } else {
+            Some(INCLUDE_GAS)
+        })
+        .with_connector_tokens(req.connector_tokens)
+        .build_query_params()
+        .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, base.decimals()))?;
+    let quote = ApiClient::new(ctx)
+        .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, base.decimals()))?
+        .get_classic_swap_quote(base.chain_id(), query_params)
+        .await
+        .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, base.decimals()))?; // use 'base' as amount in errors is in the src coin
+    ClassicSwapResponse::from_api_value(quote, rel.decimals()) // use 'rel' as quote value is in the dst coin
+        .mm_err(|err| ApiIntegrationRpcError::ApiDataError(err.to_string()))
+}
+
+/// "1inch_classic_swap_create" rpc impl
+pub async fn one_inch_v6_0_classic_swap_create_rpc(
```

Comment:
Note, that this rpc actually creates a transaction to call the swap aggregation contract. It is supposed that the GUI should sign it and send. We don't verify the tx in any way and trust the 1inch api.

Reply:
Then lest add this clarification to doc comment

Reply:
done